src/libp2p/peers.rs

```diff
@@ -1322,6 +1323,35 @@ where
         }
     }
 
+    /// Returns `true` if it is allowed to call [`Peers::queue_notification`], in other words if
+    /// there is an outbound notifications substream currently open with the target.
+    ///
+    /// If this function returns `false`, calling [`Peers::queue_notification`] will panic.
+    pub fn can_queue_notification(
+        &self,
+        target: &PeerId,
+        notifications_protocol_index: usize,
+    ) -> bool {
+        let peer_index = match self.peer_indices.get(target) {
+            Some(idx) => *idx,
+            None => return false,
+        };
+
+        match self
+            .peers_notifications_out
+            .get(&(peer_index, notifications_protocol_index))
+            .map(|state| &state.open)
+        {
+            Some(NotificationsOutOpenState::Open(_)) => true,
+            None
```

Comment:
```suggestion
            _ => false
```

?

Reply:
Yeah it's technically the same, but enumerating all the variants explicitly protects against mistakes if we add another variant to the enum.